### ISSUE TITLE
relevance sums of neuron_selection_mode="index" depend on batch size

### ISSUE THREAD

**Issue by maxkohlbrenner** [albermax/innvestigate#91]
When using neuron_selection_mode="index", the sum of the relevances increases in function of the batch size. While relevance is more-less conserved when using the top-class relevance, the relevance sum increases almost linearly with the batch sze for neuron_selection_mode="index".

![relevance_sum_ratio](https://user-images.githubusercontent.com/23200450/43966569-898fd306-9cc2-11e8-811c-b76ccf29c5c5.png)

Graphic was generated with the following script (start from the examples folder or adjust path to utils):
[keras_relevance_sums.py.tar.gz](https://github.com/albermax/innvestigate/files/2278535/keras_relevance_sums.py.tar.gz)

**Comment by enryH** [comment]
I see. I thought it is omitted for redistributing relevances and that in case of constraining it to a negative value, given that I have a value greater than 0 in a ReLU activation, I will have for sure a root point. 

From your experience: Would you included biases or rather leave them out ?

**Comment by sebastian-lapuschkin** [comment]
I see you specifically use DTD, in which case the bias would vanish due to taking the derivative (for DTD leave the bias out.)

I just had a look at the code and it seems like the DeepTaylor class inherits from the wrong LRP preset and thus treats the bias as a neuron.
Can you change its parent class from `LRPAlpha1Beta0` to `LRPAlpha1Beta0IgnoreBias` and check whether the gap between the sum of input relevances is closer to the sum of your analyzed logit of choice?
Finite precision floating point numerics will always prevent complete conservation of relevance

**Comment by albermax** [comment]
This issue should be fixed. Please check if it works as you expect.

**Comment by maxkohlbrenner** [comment]
![relevance_sum_ratio](https://user-images.githubusercontent.com/23200450/44724515-32afbf00-aad3-11e8-90c6-d964199ce76c.png)

Seems foxed, the rest is probably good up to numerical issues / fluctuations due to the bias terms.

**Comment by enryH** [comment]
In a Neural Network with ReLUs and DropOut, the scores (in logits or sigmoid case) are decomposed correctly. To summarize: Use LRPAlpha1Beta0IgnoreBias and DTD works (even for unsensible cases).

 In which paper treating the bias as a neuron is proposed? I have never heard of it!

**Comment by albermax** [comment]
Thanks for checking! I'm pretty sure this are numerical issues. Esp as it increases and decreases.

@enryH This is common praxis. A dense layer with bias is equivalent to a dense layer without and one more input that is set to the constant 1.
